.vale/styles/AsciiDoc/ValidConditions.yml

```diff
@@ -0,0 +1,26 @@
+---
+extends: script
+ignorecase: true
+level: error
+message: "Check that the if statement has a closing endif."
+link: https://docs.asciidoctor.org/asciidoc/latest/directives/ifdef-ifndef/
+scope: raw
+script: |
+  text := import("text")
+  
+  matches := []
+
+  if_regex := "(ifdef::[0-9a-zA-Z_\\-]+\\[\\]|ifndef::[0-9a-zA-Z_\\-]+\\[\\]|ifeval::\\[.*\\])"
+  endif_regex := "endif::.*"
+
+  for line in text.split(scope, "\n") {
+    if text.re_match(if_regex, line) {
+      start := text.index(scope, line)
+      matches = append(matches, {begin: start, end: start + len(line)})
+    } else if text.re_match(endif_regex, line) {
+      if len(matches) > 0 {
```

Comment:
To handle orphan endifs, maybe an additional clause `if len(matches) == 0` or something similar